app/controllers/admin/api/buyers_applications_controller.rb

```diff
@@ -95,7 +98,8 @@ def show
   #
   def update
     application.unflattened_attributes = flat_params
-    application.user_key = params[:user_key] if params[:user_key]
+    user_key_param = params[:user_key]
```

Comment:
This as well.

Reply:
This is too due to code climate issue

Reply:
I am talking about `params[:user_key]`. That's what we are fixing.

Since the `user_key` is not permitted it will produce `unable to convert unpermitted parameters to hash (ActionController::UnfilteredParameters)`

Reply:
Yes, I tried permitting all params and then require but tests failed when I did so.

def local_params
    params.permit!.to_h
  end
 
 user_key_param = local_params.require(:user_key)

Reply:
That's not correct usage. 
You can call `require` to variable which type is `ActionController::StrongParameters`.

In this case, `local_params` returns a `Hash` with all permitted parameters. We are using it because we can accept dynamic data, we might not know all the possible parameters, and so on...

In this context it would be used as: 

```
def local_params
  params.permit!.to_h
end
 
user_key_param = local_params[:user_key]
```


We are using `require` when the params MUST contain the parameter. That's why we are using it for getting the values of id's.

Reply:
Okay I got it.